src/dvc_data/hashfile/hash_info.py

```diff
@@ -1,61 +1,40 @@
-from collections import OrderedDict
-from typing import Optional
+from typing import Dict, Optional
+
+from attrs import define, field
 
 HASH_DIR_SUFFIX = ".dir"
 
 
+@define(hash=True)
 class HashInfo:
-    __slots__ = ("name", "value", "obj_name")
-
-    def __init__(
-        self,
-        name: Optional[str] = None,
-        value: Optional[str] = None,
-        obj_name: Optional[str] = None,
-    ):
-        self.name = name
-        self.value = value
-        self.obj_name = obj_name
-
-    def __eq__(self, other):
-        if not isinstance(other, HashInfo):
-            return False
-
-        return (self.name == other.name) and (self.value == other.value)
+    name: Optional[str] = None
+    value: Optional[str] = None
+    obj_name: Optional[str] = field(default=None, eq=False, hash=False)
 
     def __bool__(self) -> bool:
         return bool(self.value)
 
     def __str__(self) -> str:
         return f"{self.name}: {self.value}"
 
-    def __hash__(self) -> int:
-        return hash((self.name, self.value))
-
     @classmethod
-    def from_dict(cls, d) -> "HashInfo":
+    def from_dict(cls, d: Dict[str, str]) -> "HashInfo":
         if not d:
-            return cls(None, None)
+            return cls()
 
         ((name, value),) = d.items()
         return cls(name, value)
 
-    def to_dict(self) -> dict:
-        ret: dict = OrderedDict()
-        if not self:
-            return ret
-
-        ret[self.name] = self.value
-        return ret
+    def to_dict(self) -> Dict[str, str]:
+        if not self.value or not self.name:
+            return {}
```

Comment:
The only behavior I changed here is now if the `self.name` is None, we'll return an empty dict whereas before we would return `{None: 'value'}`. I thought this was more correct.